[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/littilemike11/Csc-448-Spring-2022/blob/main/Assignment_3.ipynb)

In [ ]:
import pandas as pd

In [ ]:
slime = pd.read_csv("https://raw.githubusercontent.com/profmcnich/example_notebook/main/science_data_large.csv")

In [ ]:
slime

Temperature °C  Mols KCL     Size nm^3
0               469       647  6.244743e+05
1               403       694  5.779610e+05
2               302       975  6.196847e+05
3               779       916  1.460449e+06
4               901        18  4.325726e+04
..              ...       ...           ...
995             894       847  1.545661e+06
996             327       982  6.737041e+05
997             791       213  3.477543e+05
998             769       553  8.684794e+05
999             919       452  8.476413e+05

[1000 rows x 3 columns]

In [ ]:
slime.head(15)

Temperature °C  Mols KCL     Size nm^3
0              469       647  6.244743e+05
1              403       694  5.779610e+05
2              302       975  6.196847e+05
3              779       916  1.460449e+06
4              901        18  4.325726e+04
5              545       637  7.124634e+05
6              660       519  7.006960e+05
7              143       869  2.718260e+05
8               89       461  8.919803e+04
9              294       776  4.770210e+05
10             991       117  2.441771e+05
11             307       781  5.006455e+05
12             206        70  3.145200e+04
13             437       599  5.390215e+05
14             566        75  9.185271e+04

In [ ]:
slime.dtypes

Temperature °C      int64
Mols KCL            int64
Size nm^3         float64
dtype: object

In [ ]:
slime.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Temperature °C  1000 non-null   int64  
 1   Mols KCL        1000 non-null   int64  
 2   Size nm^3       1000 non-null   float64
dtypes: float64(1), int64(2)
memory usage: 23.6 KB


In [ ]:
# split data into features( temp and mols kcl) and label (size) 
label = slime["Size nm^3"]
features = slime[['Temperature °C','Mols KCL' ]]



In [ ]:
# split data into test(10%) and train (90%)
import numpy as np
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(features, label, test_size = 0.10, random_state = 0)
X_train
Y_train
X_test
Y_test

993    1.705684e+05
859    2.878027e+05
298    1.577891e+06
553    4.550711e+04
672    3.558626e+04
           ...     
485    1.785532e+06
568    4.891863e+05
108    2.827965e+05
367    3.118813e+05
644    6.808943e+05
Name: Size nm^3, Length: 100, dtype: float64

In [ ]:
train_test_split(Y, shuffle = False)

In [ ]:
from sklearn import linear_model
reg= linear_model.LinearRegression()
reg.fit(X_train,Y_train)
reg.score(X_test,Y_test)



0.8761646752736477

The score is .87616 meaning that the regression line that represents this model has an accuracy of about 85%. In other words, any given predicted value on average would be about +/- 13% of the actual value. This is a fairly accurate model, as the score is quite high. The higher the score, the closer the actual score will be to the predict score.



In [ ]:
reg.predict(X_test)

array([ 134891.57109752,  566848.1770532 , 1241137.60975794,
        -24367.1696409 ,  -70657.3340248 ,  384956.44708746,
        467392.47066236,  960873.69862144,    8407.39619923,
         51611.35989487,  838231.36519831, 1232720.46674085,
        901563.3931183 , -118835.20674597,  188176.20244545,
       1158467.7057509 ,  665263.26735435,  560560.92590561,
        163940.53886357,  469055.93661764, 1419180.64745835,
        -84401.29716064,  555956.53417346, 1401489.68980634,
        450621.8221688 , 1161403.05700777,   45784.96014151,
        408948.47482721,  421868.97874553,  645969.69402768,
        467893.44392697,  208517.41869172, 1438187.73449241,
        708330.20055877, 1400047.62029314,  711132.70221441,
        942380.06884206, 1116186.8384737 , 1257396.19462681,
        189369.12027644,   71374.08771584,  870102.22808014,
         94087.37936384,  407174.28131093,  271473.01984826,
        388791.32041263,  237873.60116061,  790013.37060675,
        112125.67358907,

In [ ]:
reg.coef_

array([ 863.58108791, 1006.12741921])

In [ ]:
reg.intercept_

-400305.9133335322

Equation:
h(x,y) = 1026.657y + 879.91x - 416783.918

In [ ]:
#have issues with cross val
from sklearn.model_selection import  cross_val_score
from sklearn import svm

clf = svm.SVC(kernel='linear', C=1, random_state=50)
scores = cross_val_score(clf, features, label, cv=5)
scores

/usr/local/lib/python3.7/dist-packages/sklearn/model_selection/_validation.py:372: FitFailedWarning: 
5 fits failed out of a total of 5.
The score on these train-test partitions for these parameters will be set to nan.
If these failures are not expected, you can try to debug them by setting error_score='raise'.

Below are more details about the failures:
--------------------------------------------------------------------------------
5 fits failed with the following error:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/sklearn/model_selection/_validation.py", line 680, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py", line 199, in fit
    y = self._validate_targets(y)
  File "/usr/local/lib/python3.7/dist-packages/sklearn/svm/_base.py", line 716, in _validate_targets
    check_classification_targets(y)
  File "/usr/local/lib/python3.7/dist-packages/sklearn/utils/mu

array([nan, nan, nan, nan, nan])

In [ ]:
features.shape,label.shape

((1000, 2), (1000,))

In [ ]:
from sklearn.preprocessing import PolynomialFeatures
poly = PolynomialFeatures(2)
poly.fit_transform(X_train)

array([[1.00000e+00, 1.80000e+01, 6.10000e+01, 3.24000e+02, 1.09800e+03,
        3.72100e+03],
       [1.00000e+00, 8.26000e+02, 3.86000e+02, 6.82276e+05, 3.18836e+05,
        1.48996e+05],
       [1.00000e+00, 5.18000e+02, 7.72000e+02, 2.68324e+05, 3.99896e+05,
        5.95984e+05],
       ...,
       [1.00000e+00, 1.53000e+02, 8.01000e+02, 2.34090e+04, 1.22553e+05,
        6.41601e+05],
       [1.00000e+00, 7.26000e+02, 7.44000e+02, 5.27076e+05, 5.40144e+05,
        5.53536e+05],
       [1.00000e+00, 4.93000e+02, 9.79000e+02, 2.43049e+05, 4.82647e+05,
        9.58441e+05]])

In [ ]:
poly= PolynomialFeatures(interaction_only=True)
poly.fit_transform(X_train)

array([[1.00000e+00, 1.80000e+01, 6.10000e+01, 1.09800e+03],
       [1.00000e+00, 8.26000e+02, 3.86000e+02, 3.18836e+05],
       [1.00000e+00, 5.18000e+02, 7.72000e+02, 3.99896e+05],
       ...,
       [1.00000e+00, 1.53000e+02, 8.01000e+02, 1.22553e+05],
       [1.00000e+00, 7.26000e+02, 7.44000e+02, 5.40144e+05],
       [1.00000e+00, 4.93000e+02, 9.79000e+02, 4.82647e+05]])

In [ ]:
poly_train = poly.fit_transform(X_train)
poly_test = poly.fit_transform(X_test)

In [ ]:
reg.fit(poly_train, Y_train)
reg.score(poly_test, Y_test)

0.9999844849899112

the polynomial score is .99998. This is highly accurate and suggests a polynomial relationship between the features and label. The score suggets that on average any predicted mass is +/- 1% of the actual mass.

In [ ]:
reg.coef_

array([ 0.        , 10.81824739, 26.84454708,  2.00164481])

In [ ]:
reg.intercept_

-3748.6371021420346

Equation: h(x,y)= 0x + 10.818x^2 + 26.844y + 2.0016y^2 - 3748.6

In [ ]:
reg.predict(poly_test)

array([ 171695.19986458,  286448.78662537, 1577700.57927154,
         47353.50137477,   37038.46017273,  319978.88678533,
        322742.75736385, 1075679.67362661,   54238.26150176,
         98793.53349122,  810316.41358459, 1563098.66853368,
        993996.00175317,   47409.73117431,  150873.62250799,
       1418720.29160849,  512957.08917505,  383622.19133419,
         89357.65415322,  402834.70255467, 1921436.96509693,
         56202.23971639,  227105.91090013, 1884276.93363533,
        413095.8372107 , 1431493.94981139,   64133.56853313,
        336688.83379537,  360677.76427022,  579065.25614379,
        433769.01212112,  113497.54090352, 1967473.33097835,
        725720.11316563, 1880155.9548512 ,  665546.13894882,
       1043763.44252199, 1346125.49545279, 1600903.87354366,
         10485.10679753,   71280.77377168,  668906.44668846,
         77540.79565159,  367599.41846858,  265967.4205479 ,
        236929.89352372,  139323.90794548,  717130.5290997 ,
         79701.15766303,

In [ ]:
Y_test

993    1.705684e+05
859    2.878027e+05
298    1.577891e+06
553    4.550711e+04
672    3.558626e+04
           ...     
485    1.785532e+06
568    4.891863e+05
108    2.827965e+05
367    3.118813e+05
644    6.808943e+05
Name: Size nm^3, Length: 100, dtype: float64